src/components/SearchBox.vue

```diff
@@ -1,75 +1,235 @@
+      searchBoxUpdate(variable) {
+        this.timeTo = variable[1];
+        this.timeFrom = variable[0];
+      },
+      //check the item that is removed from te fliters
+      //if it's a time filter, set the time on the appropriate variable to the max or min limit, depending on which filter.
+      timeSliderValuesUpdated(item) {
+        item.includes("ff_primaryobject_year_from_i") ? this.timeFrom = this.searchOptions.timeOptions.min : null
+        item.includes("ff_primaryobject_year_to_i") ? this.timeTo = this.searchOptions.timeOptions.max : null
+      },
+      //Search function h as been substancially bigger - because we have to take care of some time-issues.
       search(e) {
-        if (this.searchQuery !== this.queryDisplay) {
-          this.updateQueryDisplay(this.searchQuery);
-          let filters = this.$router.history.current.params.query;
-          let fixedFilters = "";
-          if (filters != undefined) {
-            if (filters.indexOf("&d=") > -1 || filters.indexOf("&fq=") > -1) {
-              if (filters.indexOf("&d=") > filters.indexOf("&fq=")) {
-                fixedFilters = "&d=" + filters.split("&d=").pop();
-              } else {
-                fixedFilters = "&fq=" + filters.split("&fq=").pop();
-              }
+        // We check if anything has changed. Either the query or any time related business.
+        if (this.searchQuery !== this.queryDisplay || this.queryTimeChanged()) {
+          // If we're satisfied, we check if the searchQuery is set. If it isn't, we set it to all
+          // This is if you're only searching based on time.
+          this.searchQuery === '' ? this.searchQuery = "*.*" : null;
+          this.updateQueryDisplay(this.searchQuery || "*.*");
+          //Rip out the filters from he params and set up variables for creating new filter.
+          let filters = this.$router.history.current.params.query || '';
+          let newCombinedFilter = "";
+          let filterArray = [];
+          // Make sure there actually are some filters to run manipulate.
+          if (filters !== undefined) {
+            // Check if the first filter is a normal filter or a location filter.
+            let isLocationSet = filters.indexOf("&d=") > -1;
+            // split the filter up accordingly - if location, we split on &d= first, then on &fq=.
+            isLocationSet ? filterArray = filters.split("&d=")[1].split("&fq=") : filterArray = filters.split("&fq=");
+            // If location is not set, we hae to remove the first entry, as it's the query.
+            // If location is true, then we've alread done it with the [1] in the filter.split above.
+            isLocationSet ? null : filterArray.shift();
+            //Run through the array, and inset the new updated filters - but make sure we actually have an array to run through.
+            if(filterArray.length > 0 && Array.isArray(filterArray)) {
```

Comment:
I had no idea! How do you know stuff like this? I'm doing the check because I encountered times where the manipulation of the variable with .pop, .shift etc could result in a string. I needed to make sure I never entered the loop if it wasn't an array. I'll change the order tho.